### PR TITLE
ed: extrapolation of house numbers in geopal

### PR DESCRIPTION
**Pull request by TeXitoi**
fix http://jira.canaltp.fr/browse/NAVITIAII-1406

### CODE REVIEW

File: source/ed/connectors/geopal_parser.cpp [hunk 1]
```diff
@@ -156,6 +155,146 @@ void GeopalParser::fill_admins(){
...
+// 2                       8
+//
+// and we generate something like that:
+//
+//    1     3     6     9
+//    +     +     +     +
+// +-----------------------+
+//     +       +       +
+//     2       4       8
+struct HouseNumberFromEdgesFiller {
```
Comment (l-vincent-l) [inline source/ed/connectors/geopal_parser.cpp:173]: Some unit_tests would be nice.

File: source/ed/connectors/geopal_parser.cpp [hunk 2]
```diff
@@ -156,6 +155,146 @@ void GeopalParser::fill_admins(){
...
+        const auto vec = get_trigo_ortho(to, from);
+        add_middle(translate(from, vec), translate(to, vec), begin_num, last_num, way);
+    }
+    void add_middle(const Coord& from,
+                    const Coord& to,
+                    const int begin_num,
+                    const int last_num,
+                    Way* way) {
+        const double nb = double((last_num - begin_num) / 2 + 1);
+        Coord vec((to.lon() - from.lon()) / nb / 2, (to.lat() - from.lat()) / nb / 2);
```
Comment (l-vincent-l) [inline source/ed/connectors/geopal_parser.cpp]: is vec a vector ?

File: source/ed/connectors/data_cleaner.cpp [hunk 3]
```diff
@@ -64,61 +64,36 @@ void data_cleaner::fusion_ways() {
 
         way_vector.insert(way_vector.begin(), way.second->edges.begin(),  way.second->edges.end());
     }
+    std::map<types::Way*, types::Way*> traduction;
```
Comment (l-vincent-l) [inline source/ed/connectors/data_cleaner.cpp]: please, call it translations.

File: source/ed/connectors/geopal_parser.cpp (hunk 2 above)
Comment (TeXitoi) [inline source/ed/connectors/geopal_parser.cpp]: it's a geometric vector

File: source/ed/connectors/geopal_parser.cpp (hunk 1 above)
Comment (TeXitoi) [inline source/ed/connectors/geopal_parser.cpp:173]: On geopal parsing??? and how do you want to test that?

File: source/ed/connectors/geopal_parser.cpp [hunk 4]
```diff
@@ -156,6 +155,167 @@ void GeopalParser::fill_admins(){
...
+// generate something like that:
+//
+//    1     3     6     9
+//    +     +     +     +
+// +----------------------->
+//     +       +       +
+//     2       4       8
+struct HouseNumberFromEdgesFiller {
+    typedef navitia::type::GeographicalCoord Coord;
+    typedef const std::vector<std::string>& Row;
```
Comment (antoine-de) [inline source/ed/connectors/geopal_parser.cpp]: hum I find that a bit more clear not to put the & stuff on the typedef but at each use no ?

**Comment by antoine-de** [comment]
ok for me

**Pull request merged**